services/hierarchy/src/main/kotlin/IssueService.kt

```diff
@@ -0,0 +1,228 @@
+
+        val paginatedQuery = parameters.limit?.let { orderedQuery.limit(it, parameters.offset ?: 0) }
+            ?: orderedQuery
+
+        ListQueryResult(
+            paginatedQuery.map(ResultRow::toIssueWithIdentifier),
+            parameters,
+            combinedQuery.count()
+        )
+    }
+
+    private fun createOrtRunIssuesQuery(ortRunId: Long) = (IssuesTable innerJoin OrtRunsIssuesTable).select(
+        IssuesTable.timestamp,
+        IssuesTable.issueSource,
+        IssuesTable.message,
+        IssuesTable.severity,
+        IssuesTable.affectedPath,
+        Op.nullOp<Unit>().alias("\"type\""),
+        Op.nullOp<Unit>().alias("\"namespace\""),
+        Op.nullOp<Unit>().alias("\"name\""),
+        Op.nullOp<Unit>().alias("\"version\"")
+    ).where { OrtRunsIssuesTable.ortRunId eq ortRunId }
+
+    private fun createAnalyzerIssuesQuery(ortRunId: Long): Query {
+        val analyzerIssuesIdentifiersJoin = (
+                IssuesTable innerJoin
+                        IdentifiersIssuesTable innerJoin
+                        AnalyzerRunsIdentifiersIssuesTable innerJoin
+                        AnalyzerRunsTable innerJoin
+                        AnalyzerJobsTable innerJoin
+                        IdentifiersTable
+                )
+
+        return analyzerIssuesIdentifiersJoin.select(
+            IssuesTable.timestamp,
+            IssuesTable.issueSource,
+            IssuesTable.message,
+            IssuesTable.severity,
+            IssuesTable.affectedPath,
+            IdentifiersTable.type,
+            IdentifiersTable.name,
+            IdentifiersTable.namespace,
+            IdentifiersTable.version
+        ).where { AnalyzerJobsTable.ortRunId eq ortRunId }
+    }
+
+    private fun createAdvisorIssuesQuery(ortRunId: Long): Query {
+        val advisorIssuesIdentifiersJoin = IssuesTable
+            .innerJoin(AdvisorResultsIssuesTable, { IssuesTable.id }, { issueId })
+            .innerJoin(AdvisorResultsTable, { AdvisorResultsIssuesTable.advisorResultId }, { AdvisorResultsTable.id })
+            .innerJoin(AdvisorRunsTable, { AdvisorResultsTable.advisorRunIdentifierId }, { AdvisorRunsTable.id })
+            .innerJoin(AdvisorJobsTable, { AdvisorRunsTable.advisorJobId }, { AdvisorJobsTable.id })
+            .innerJoin(AdvisorRunsIdentifiersTable, { AdvisorRunsTable.id }, { advisorRunId })
+            .innerJoin(IdentifiersTable, { AdvisorRunsIdentifiersTable.identifierId }, { IdentifiersTable.id })
+
+        return advisorIssuesIdentifiersJoin.select(
+            IssuesTable.timestamp,
+            IssuesTable.issueSource,
+            IssuesTable.message,
+            IssuesTable.severity,
+            IssuesTable.affectedPath,
+            IdentifiersTable.type,
+            IdentifiersTable.name,
+            IdentifiersTable.namespace,
+            IdentifiersTable.version
+        ).where { AdvisorJobsTable.ortRunId eq ortRunId }
+    }
+
+    /*
```

Comment:
Not sure whether this out-commented code should remain here, since it needs to be reworked anyway. Maybe better add a TODO comment (to the class) describing the current limitations.

Reply:
Removed the commented out code and add a TODO with what is missing.